Project/Assets/ML-Agents/Examples/Walker/Scripts/WalkerAgent.cs

```diff
@@ -9,14 +9,32 @@
 
 public class WalkerAgent : Agent
 {
-    public float maximumWalkingSpeed = 999; //The max walk velocity magnitude an agent will be rewarded for
-    Vector3 m_WalkDir; //Direction to the target
-//    Quaternion m_WalkDirLookRot; //Will hold the rotation to our target
+    [Header("Walk Speed")] [Range(0.1f, 10)] public float targetWalkingSpeed = 10; //The walking speed to try and achieve
```

Comment:
This field is public, so nothing is preventing the user to call `targetWalkingSpeed =0f` via code and raise the NaNs. I think this property should be private (like a lot of other properties).

Reply:
agreed. on the NaN issue. changed to use custom getters/setters. custom setter now clamps between (.1, m_maxWalkingSpeed). changed targetWalkingSpeed to be private